src/components/MapStyleInput.svelte

```diff
@@ -212,7 +214,12 @@
         break;
       }
       case 'branch': {
-        textInput = '';
+        if (isUrl(textInput)) {
```

Comment:
This mostly works, but we might want to handle 2 more scenarios:
- When someone types in `locahost:8000...` we have [special handling](https://github.com/stamen/map-compare/blob/79f8603d19907512df77d9dab70af364f06f9a9b/src/components/MapStyleInput.svelte#L168-L172) so they don't need to add `http://`. We should account for this since currently a localhost url without the preface will persist when switching to a branch pattern
- Filling out an invalid or incomplete custom URL (so long as it doesn't contain `://`) then switching to a branch pattern will persist that text input. We might want to check in addition (or instead) if the current url matches the text input here instead of just checking that it is a url